core/src/main/scala/com/avast/grpc/jsonbridge/ReflectionGrpcJsonBridge.scala

```diff
@@ -2,93 +2,105 @@ package com.avast.grpc.jsonbridge
 
 import java.lang.reflect.Method
 
-import cats.effect.Async
-import cats.syntax.all._
+import cats.effect._
+import cats.implicits._
 import com.avast.grpc.jsonbridge.GrpcJsonBridge.GrpcMethodName
-import com.google.common.util.concurrent._
+import com.google.common.util.concurrent.{FutureCallback, Futures, ListenableFuture}
 import com.google.protobuf.util.JsonFormat
 import com.google.protobuf.{Message, MessageOrBuilder}
 import com.typesafe.scalalogging.StrictLogging
 import io.grpc.MethodDescriptor.{MethodType, PrototypeMarshaller}
-import io.grpc._
 import io.grpc.inprocess.{InProcessChannelBuilder, InProcessServerBuilder}
+import io.grpc._
 import io.grpc.stub.AbstractStub
 
 import scala.collection.JavaConverters._
 import scala.concurrent.ExecutionContext
-import scala.language.{existentials, higherKinds}
+import scala.language.existentials
+import scala.language.higherKinds
 import scala.util.control.NonFatal
 
-class ReflectionGrpcJsonBridge[F[_]](services: ServerServiceDefinition*)(implicit ec: ExecutionContext, F: Async[F])
-    extends GrpcJsonBridge[F]
-    with AutoCloseable
-    with StrictLogging {
-
-  import com.avast.grpc.jsonbridge.ReflectionGrpcJsonBridge._
+object ReflectionGrpcJsonBridge extends StrictLogging {
 
-  def this(grpcServer: io.grpc.Server)(implicit ec: ExecutionContext, F: Async[F]) = this(grpcServer.getImmutableServices.asScala: _*)
+  // JSON body and headers to a response (fail status or JSON response)
+  private type HandlerFunc[F[_]] = (String, Map[String, String]) => F[Either[Status, String]]
 
-  private val inProcessServiceName = s"HttpWrapper-${System.nanoTime()}"
+  private val parser: JsonFormat.Parser = JsonFormat.parser()
 
-  private val inProcessServer = {
-    val b = InProcessServerBuilder.forName(inProcessServiceName).executor(ec.execute(_))
-    services.foreach(b.addService)
-    b.build().start()
+  private val printer: JsonFormat.Printer = {
+    JsonFormat.printer().includingDefaultValueFields().omittingInsignificantWhitespace()
   }
 
-  private val inProcessChannel = InProcessChannelBuilder.forName(inProcessServiceName).executor(ec.execute(_)).build()
+  def createFromServer[F[_]](ec: ExecutionContext)(grpcServer: io.grpc.Server)(implicit F: Async[F]): Resource[F, GrpcJsonBridge[F]] = {
+    createFromServices(ec)(grpcServer.getImmutableServices.asScala: _*)
+  }
 
-  override def close(): Unit = {
-    inProcessChannel.shutdownNow()
-    inProcessServer.shutdownNow()
-    ()
+  def createFromServices[F[_]](ec: ExecutionContext)(services: ServerServiceDefinition*)(
+      implicit F: Async[F]): Resource[F, GrpcJsonBridge[F]] = {
+    val inProcessServiceName = s"HttpWrapper-${System.nanoTime()}"
```

Comment:
I believe this has to be inside the `F`.

Reply:
weeeeel, it would work quite fine
but it's dirty, so OK, I've suspended it ;)

Reply:
It would not - you need the name to be unique for each execution, which it would not in this case.